src/Footer/Component.tsx

```diff
@@ -1,29 +1,72 @@
+import { ImageMedia } from '@/components/Media/ImageMedia'
+import { Icons } from '@/components/ui/icons'
+import { getPayload } from 'payload'
 
 export async function Footer({ center }: { center?: string }) {
-  const footerData: Footer = await getCachedGlobal('footer', 1)()
-
-  const navItems = footerData?.navItems || []
-
+  const payload = await getPayload({ config: configPromise })
+  const { docs: data } = await payload.find({
+    collection: 'footer',
+    depth: 10,
+    where: {
+      'tenant.slug': {
+        equals: center,
+      },
+    },
+  })
+  const { address, email, hashtag, logo, name, phone, privacy, socialMedia, terms } = data[0]
   return (
     <footer className="mt-auto border-t border-border bg-footer text-footer-foreground">
-      <div className="container py-8 gap-8 flex flex-col md:flex-row md:justify-between">
-        <Link className="flex items-center" href="/">
-          <Logo center={center} />
-        </Link>
-
-        <div className="flex flex-col-reverse items-start md:flex-row gap-4 md:items-center">
-          <nav className="flex flex-col md:flex-row gap-4">
-            {navItems.map(({ link }, i) => {
-              return <CMSLink className="text-footer-foreground" key={i} {...link} />
-            })}
-          </nav>
+      <div className="container py-8 gap-8 grid grid-cols-3">
+        <div>
+          <h4 className="font-medium text-xl mb-2">Stay updated!</h4>
+          <p>Sign up for our newsletter</p>
+        </div>
+        <div>
+          <Link className="flex items-center" href="/">
+            {logo ? <ImageMedia resource={logo} /> : <Logo center={center} />}
+          </Link>
+        </div>
+        <div>
+          <div className="flex flex-col items-start">
+            <h4 className="font-medium text-xl mb-2">{name}</h4>
+            {address && <div className="whitespace-pre-line">{address}</div>}
+            {phone && <a href={`tel:${phone}`}>{phone}</a>}
+            {email && (
+              <a className="mb-6 text-secondary underline" href={`mailto:${email}`}>
+                {email}
+              </a>
+            )}
+            {socialMedia && (
+              <div className="flex gap-x-4 mb-2">
+                {socialMedia.instagram && <Icons.instagram />}
+                {socialMedia.facebook && <Icons.facebook />}
+                {socialMedia.twitter && <Icons.twitter />}
+                {socialMedia.linkedin && <Icons.linkedin />}
+                {socialMedia.youtube && <Icons.youtube />}
+              </div>
+            )}
+            {hashtag && <p className="mb-4 text-secondary underline">{hashtag}</p>}
+          </div>
+        </div>
+      </div>
+      <div className="container text-center pb-8">
+        <p className="mb-2">All Content © 2017 – 2025 {name}</p>
+        <div className="flex gap-x-2 justify-center">
+          {typeof terms === 'object' && (
+            <a href={`/${terms?.slug}`} className="underline">
```

Comment:
`terms` is a required reference to a page - why do we need the `?`? In what case would it be `null`?

Reply:
I have this as optional since I am not sure every AC has Terms & Conditions and a Privacy policy. I switched to have an initial null check rather than inline.